app/channel.go

```diff
@@ -1370,7 +1370,7 @@ func (a *App) addUserToChannel(user *model.User, channel *model.Channel, teamMem
 }
 
 func (a *App) AddUserToChannel(user *model.User, channel *model.Channel) (*model.ChannelMember, *model.AppError) {
-	teamMember, nErr := a.Srv().Store.Team().GetMember(channel.TeamId, user.Id)
+	teamMember, nErr := a.Srv().Store.Team().GetMember(channel.TeamId, user.Id, true)
```

Comment:
This change is not localized to just LDAP sync, but I'm assuming the performance impact will be low.